### PR TITLE
Add "Manage custom filters" context menu item

### PR DESCRIPTION
**Pull request by antonok-edm**
Resolves https://github.com/brave/brave-browser/issues/14978

I've also added 2 commits here to clean up minor details from https://github.com/brave/brave-core/pull/8156 - one function no longer needs to be async and there are a bunch of locale strings that can now be removed safely.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

From any page, right click on the page and check that the `Brave > Manage custom filters` context menu item exists and navigates to a new `brave://adblock` tab, or an existing one if present.

### CODE REVIEW

File: components/brave_extension/extension/brave_extension/background/api/cosmeticFilterAPI.ts [hunk 1]
```diff
@@ -94,10 +94,14 @@ const tryMigratingLegacyCosmeticFilters = () => {
   })
 }
 
-export const addSiteCosmeticFilter = async (host: string, cssSelector: string) => {
+export const addSiteCosmeticFilter = (host: string, cssSelector: string) => {
```
Comment (bsclifton) [inline components/brave_extension/extension/brave_extension/background/api/cosmeticFilterAPI.ts:97]: Should this be changing to not async? I know there is a delay when trying to calculate the selector sometimes and I was able to get it to lag the browser quite a bit by doing right click / adding a lot of selectors on a news site

File: components/brave_extension/extension/brave_extension/background/events/cosmeticFilterEvents.ts [hunk 2]
```diff
@@ -89,7 +98,7 @@ export function tabsCallback (tabs: any) {
   chrome.tabs.sendMessage(tabs[0].id, { type: 'getTargetSelector' }, onSelectorReturned)
 }
 
-export async function onSelectorReturned (response: any) {
+export function onSelectorReturned (response: any) {
```
Comment (bsclifton) [inline components/brave_extension/extension/brave_extension/background/events/cosmeticFilterEvents.ts:101]: same question about should it be removed as async?

**Review by bsclifton — COMMENTED** [review]
Functionality works great! Just had a question about the async keyword removal on `addSiteCosmeticFilter`. The other changes look good! 😄

**Comment by antonok-edm** [comment]
@bsclifton after discussion with @pes10k it looks like the `async` change shouldn't affect anything - `onSelectorReturned` is only used as the callback in `chrome.tabs.sendMessage(tabs[0].id, { type: 'getTargetSelector' }, onSelectorReturned)`, and the return value of that callback is ignored as per https://developer.chrome.com/docs/extensions/reference/tabs/#method-sendMessage

**Review by bsclifton — APPROVED** [review]
@antonok-edm perfect - thanks for confirming. Sorry I didn't get to dig in there more after raising the comment. LGTM! 😄👍

**Comment by bbondy** [comment]
Did this possibly introduce this CI failure on unittests @antonok-edm ?
https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-backup-search-network-pre-init/detail/pr-brave-browser-backup-search-network-pre-init/9/pipeline/

**Comment by antonok-edm** [comment]
Yep, my fault - sorry about that! :grimacing: 

I just opened https://github.com/brave/brave-core/pull/8429 to update the tests accordingly,

**Comment by bsclifton** [comment]
ah crap - my bad on that 😱 I saw the pre-init failure and (unfortunately) assumed it was an npm vulnerability. Great catch @bbondy 😄

**Pull request merged**